jax/experimental/sparse_ops.py

```diff
@@ -51,11 +51,52 @@
 from jax.interpreters import ad
 from jax.util import safe_zip
 from jax._src.lax.lax import ranges_like, remaining, _dot_general_batch_dim_nums
+from jax.api_util import _ensure_index, _ensure_index_tuple
 
 xb = xla_bridge
 xops = xla_client.ops
 Dtype = Any
 
+
+def grad(fun: Callable, argnums: Union[int, Sequence[int]] = 0, **kwargs) -> Callable:
```

Comment:
Maybe worth doing `value_and_grad`, and making `grad` a wrapper of that, as is the case in `api.py`? In my experience, it is pretty common to want to pull out the `loss` when doing gradient based optimization. (and it doesn't add any additional cost)

Reply:
Good idea - Done!